INFO:root:Вызвана страница всех клиентов
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\tickets\views.py changed, reloading.
INFO:root:Вызвана страница всех промокодов
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\rooms\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница всех клиентов
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница всех работников
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\urls.py changed, reloading.
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех клиентов
INFO:root:Юзер None удален
INFO:root:Вызвана страница всех клиентов
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана функция поиска работников
ERROR:django.request:Internal Server Error: /employees/search-result/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\views.py", line 30, in search_employees
    employees = Employee.objects.filter(name__icontains=search_query)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1495, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1307, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1772, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'name' into field. Choices are: age, animal, id, image, phone_number, position, position_id, room, room_id, time_create, time_update, user, user_id
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\views.py changed, reloading.
INFO:root:Вызвана функция поиска работников
ERROR:django.request:Internal Server Error: /employees/search-result/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\views.py", line 30, in search_employees
    employees = Employee.objects.filter(room__icontains=search_query)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1559, in build_filter
    condition = self.build_lookup(lookups, col, value)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1383, in build_lookup
    lhs = self.try_transform(lhs, lookup_name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1427, in try_transform
    raise FieldError(
django.core.exceptions.FieldError: Unsupported lookup 'icontains' for ForeignKey or join on the field not permitted.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\views.py changed, reloading.
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана функция поиска работников
INFO:root:Вызвана функция поиска работников
INFO:root:Вызвана функция поиска работников
INFO:root:Вызвана функция поиска работников
INFO:root:Вызвана функция поиска работников
INFO:root:Вызвана функция поиска работников
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\users\forms.py changed, reloading.
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех клиентов
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 29 вошел
INFO:root:Вызвана страница всех животных
WARNING:root:Form isnt valid
INFO:root:Юзер 29 изменен
INFO:root:Вызвана страница всех животных
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 25 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 25 изменен
INFO:root:Вызвана страница всех животных
WARNING:root:Form isnt valid
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех клиентов
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех клиентов
INFO:root:Юзер None удален
INFO:root:Вызвана страница всех клиентов
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 8 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 8 изменен
INFO:root:Вызвана страница всех животных
WARNING:root:Form isnt valid
INFO:root:Юзер 8 изменен
INFO:root:Вызвана страница всех животных
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\users\services\agebyname.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\users\services\agebyname.py changed, reloading.
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница животных работника
INFO:root:Вызвана страница животных работника
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех клиентов
INFO:root:Вызвана главная страница
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница контакты
INFO:root:Вызвана главная страница
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\models.py changed, reloading.
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех работников
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\employees\forms.py changed, reloading.
INFO:root:Вызвана страница всех работников
WARNING:root:Form isnt valid
INFO:root:Работник 2 изменен
INFO:root:Вызвана страница всех работников
INFO:root:Работник 4 изменен
INFO:root:Вызвана страница всех работников
INFO:root:Работник 1 изменен
INFO:root:Вызвана страница всех работников
INFO:root:Работник 6 изменен
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех животных
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех работников
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница контакты
INFO:root:Вызвана страница контакты
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница всех промокодов
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех промокодов
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:root:Юзер 29 вошел
INFO:root:Вызвана страница всех животных
ERROR:django.request:Internal Server Error: /add_review
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py", line 121, in add_review
    return render(request, 'reviews/add_review.html', {'form': form, 'us_gr': us_gr})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: reviews/add_review.html
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:root:Вызвана страница отзывов
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница отзывов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана главная страница
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница отзывов
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:root:Вызвана страница о сайте
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Вызвана страница всех помещений
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Вызвана страница всех помещений
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
WARNING:django.request:Forbidden (Permission denied): /animals/add/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 22, in _wrapper_view
    if test_func(request.user):
       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 78, in check_perms
    raise PermissionDenied
django.core.exceptions.PermissionDenied
INFO:root:Комната добавлена
INFO:root:Комната добавлена
INFO:root:Комната 1 удалена
INFO:root:Вызвана страница всех помещений
INFO:root:Юзер None вышел
INFO:root:Новый промокод добавлен
INFO:root:Новый промокод добавлен
INFO:root:Код TESTCODE удален
INFO:root:Промокод изменен
INFO:root:Вызвана страница о сайте
WARNING:django.request:Not Found: /
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница отзывов
ERROR:django.request:Internal Server Error: /reviews
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py", line 102, in allreviews
    return render(request, 'mainapp/reviews.html', {'reviews': reviews, 'us_gr': us_gr})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 326, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 326, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'del_code' with keyword arguments '{'code_id': ''}' not found. 1 pattern(s) tried: ['tickets/delete\\-code/(?P<code_id>[0-9]+)\\Z']
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:root:Вызвана страница отзывов
ERROR:django.request:Internal Server Error: /reviews
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py", line 102, in allreviews
    return render(request, 'mainapp/reviews.html', {'reviews': reviews, 'us_gr': us_gr})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 326, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 326, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\template\defaulttags.py", line 479, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'delete_review' with keyword arguments '{'id': 1}' not found. 1 pattern(s) tried: ['delete_review/<int:id>']
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:root:Вызвана страница отзывов
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница отзывов
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница всех клиентов
INFO:root:Вызвана страница всех промокодов
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 8 вошел
INFO:root:Вызвана страница всех животных
WARNING:django.request:Not Found: /
WARNING:django.request:Not Found: /
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\urls.py changed, reloading.
INFO:root:Вызвана главная страница
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница отзывов
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница животных работника
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана страница животных работника
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 29 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана главная страница
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница отзывов
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана главная страница
WARNING:django.request:Not Found: /favicon.ico
INFO:root:Вызвана главная страница
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница всех промокодов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница отзывов
INFO:root:Вызвана страница отзывов
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:django.utils.autoreload:C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\mainapp\views.py changed, reloading.
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница о сайте
INFO:root:Вызвана страница новостей
INFO:root:Вызвана страница всех животных
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 29 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Билет 35 куплен zodi0204
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана функция поиска
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана главная страница
INFO:root:Вызвана главная страница
INFO:root:Вызвана главная страница
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех работников
INFO:root:Работник удален
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Комната 13 изменена
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех помещений
INFO:root:Комната добавлена
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница всех работников
INFO:root:Работник удален
INFO:root:Вызвана страница всех работников
INFO:root:Работник удален
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех работников
INFO:root:Работник удален
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех работников
WARNING:root:Form isnt valid
INFO:root:Работник 10 изменен
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех животных
INFO:root:Животное добавлено
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех животных
INFO:root:Животное добавлено
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница всех животных
INFO:root:Животное изменено
INFO:root:Вызвана страница всех животных
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 42 вошел
INFO:root:Вызвана страница всех животных
WARNING:root:Form isnt valid
INFO:root:Юзер 42 изменен
INFO:root:Вызвана страница всех животных
ERROR:django.request:Internal Server Error: /user/profile/
Traceback (most recent call last):
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrei\PycharmProjects\web-zoo\myzoo\users\views.py", line 87, in user_profile
    return render(request, 'users/profile.html', {'form': form, 'customer': usr, 'groups': groups})
                                                                            ^^^
UnboundLocalError: cannot access local variable 'usr' where it is not associated with a value
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница всех клиентов
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана главная страница
INFO:root:Вызвана главная страница
INFO:root:Вызвана главная страница
INFO:root:Вызвана страница всех клиентов
INFO:root:Юзер None удален
INFO:root:Вызвана страница всех клиентов
INFO:root:Вызвана страница всех помещений
INFO:root:Комната добавлена
INFO:root:Вызвана страница всех помещений
INFO:root:Комната 19 изменена
INFO:root:Вызвана страница всех помещений
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница популяции по семейству
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница всех работников
INFO:root:Вызвана страница отзывов
INFO:root:Отзыв 2 добавлен
INFO:root:Вызвана страница отзывов
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 29 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Билет 36 куплен zodi0204
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана страница билетов юзера
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана функция сортировки
INFO:root:Вызвана функция поиска
INFO:root:Вызвана функция поиска
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 1 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана главная страница
WARNING:django.request:Not Found: /favicon.ico
INFO:root:Юзер None вышел
INFO:root:Вызвана страница всех животных
INFO:root:Юзер 8 вошел
INFO:root:Вызвана страница всех животных
INFO:root:Вызвана страница животных работника
